src/platform/OpenThread/GenericThreadBorderRouterDelegate.cpp

```diff
@@ -0,0 +1,207 @@
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    otError otErr = OT_ERROR_NONE;
+    {
+        ScopedThreadLock threadLock;
+        otErr = otBorderAgentGetId(otInst, &borderAgentId);
+    }
+    if (otErr == OT_ERROR_NONE)
+    {
+        CopySpanToMutableSpan(ByteSpan(borderAgentId.mId), borderAgentIdSpan);
+        return CHIP_NO_ERROR;
+    }
+    return DeviceLayer::Internal::MapOpenThreadError(otErr);
+}
+
+uint16_t GenericOpenThreadBorderRouterDelegate::GetThreadVersion()
+{
+    return otThreadGetVersion();
+}
+
+bool GenericOpenThreadBorderRouterDelegate::GetInterfaceEnabled()
+{
+    otInstance * otInst = DeviceLayer::ThreadStackMgrImpl().OTInstance();
+    VerifyOrReturnValue(otInst, false);
+    ScopedThreadLock threadLock;
+    return otIp6IsEnabled(otInst);
+}
+
+CHIP_ERROR GenericOpenThreadBorderRouterDelegate::GetDataset(Thread::OperationalDataset & dataset, DatasetType type)
+{
+    otInstance * otInst = DeviceLayer::ThreadStackMgrImpl().OTInstance();
+    VerifyOrReturnError(otInst, CHIP_ERROR_INCORRECT_STATE);
+
+    otError otErr = OT_ERROR_NONE;
+    otOperationalDatasetTlvs datasetTlvs;
+    {
+        ScopedThreadLock threadLock;
+        if (type == DatasetType::kActive)
+        {
+            otErr = otDatasetGetActiveTlvs(otInst, &datasetTlvs);
+        }
+        else
+        {
+            otErr = otDatasetGetPendingTlvs(otInst, &datasetTlvs);
+        }
+    }
+    if (otErr == OT_ERROR_NONE)
+    {
+        return dataset.Init(ByteSpan(datasetTlvs.mTlvs, datasetTlvs.mLength));
+    }
+    return DeviceLayer::Internal::MapOpenThreadError(otErr);
+}
+
+void GenericOpenThreadBorderRouterDelegate::SetActiveDataset(const Thread::OperationalDataset & activeDataset, uint32_t sequenceNum,
+                                                             ActivateDatasetCallback * callback)
+{
+    // This function will never be invoked when there is an Active Dataset already configured.
+    CHIP_ERROR err = SaveActiveDatasetConfigured(false);
+    if (err == CHIP_NO_ERROR)
+    {
+        err = DeviceLayer::ThreadStackMgrImpl().AttachToThreadNetwork(activeDataset, nullptr);
+    }
+    if (err != CHIP_NO_ERROR)
+    {
+        callback->OnActivateDatasetComplete(sequenceNum, err);
+        return;
+    }
+    mSequenceNum = sequenceNum;
+    mCallback    = callback;
+}
+
+void GenericOpenThreadBorderRouterDelegate::OnPlatformEventHandler(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    GenericOpenThreadBorderRouterDelegate * delegate = reinterpret_cast<GenericOpenThreadBorderRouterDelegate *>(arg);
+    if (delegate && delegate->mCallback)
+    {
+        if (event->Type == DeviceLayer::DeviceEventType::kThreadConnectivityChange &&
+            event->ThreadConnectivityChange.Result == DeviceLayer::kConnectivity_Established)
+        {
+            delegate->mCallback->OnActivateDatasetComplete(delegate->mSequenceNum, CHIP_NO_ERROR);
+            delegate->SaveActiveDatasetConfigured(true);
```

Comment:
If I understand correctly the latest updates, this line should be removed and instead this operation will happen in the server instance in function `ServerInstance::OnPlatformEventHandle` :

`else if (event->Type == DeviceLayer::DeviceEventType::kCommissioningComplete)
{
    _this->mDelegate->CommitActiveDataset();
}`

Reply:
Oh, I forget to remove this line. Removed it now.